components/tidb_query_vec_executors/src/table_scan_executor.rs

```diff
@@ -291,21 +295,38 @@ impl ScanExecutorImpl for TableScanExecutorImpl {
     ) -> Result<()> {
         use tidb_query_datatype::codec::{datum, table};
 
-        check_record_key(&key)?;
+        table::check_record_key(&key)?;
```

Comment:
no need to check record key for twice?

Reply:
Yes! We could move this check to a `else` arm.